device/transport/amqp/devdoc/amqp_twin_client_requirements.md

```diff
@@ -3,106 +3,139 @@
 ## Overview
 Object used to subscribe to the Cloud-to-Device messages for Twin
 
-## Example
-```javascript
-var receiver = new AmqpTwinClient(config: ClientConfig, client: any);
-receiver.on('response', function(response) {
-    console.log('Response received for request ' + response.requestId);
-    console.log('  status = " + response.status);
-    console.log(response.body);
-});
+## Public API
+
+```typescript
+class AmqpTwinClient extends EventEmitter {
+  constructor(authenticationProvider: AuthenticationProvider, client: any) {
+  getTwin(callback: (err: Error, twin?: TwinProperties) => void): void;
+  updateTwinReportedProperties(patch: any, callback: (err?: Error) => void): void;
+  enableTwinDesiredPropertiesUpdates(callback: (err?: Error) => void): void;
+  disableTwinDesiredPropertiesUpdates(callback: (err?: Error) => void): void;
+  detach(callback: (err?: Error) => void): void;
+}
 ```
+### constructor(authenticationProvider: AuthenticationProvider, client: any) {
 
-## Public API
+**SRS_NODE_DEVICE_AMQP_TWIN_06_005: [** The `AmqpDeviceMethodClient` shall inherit from the `EventEmitter` class. **]**
 
-### Constructor
+### getTwin(callback: (err: Error, twin?: TwinProperties) => void): void;
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_004: [** The `AmqpTwinClient` constructor shall throw `ReferenceError` if the `config` object is falsy. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_007: [** The `getTwin` method shall attach the sender link if it's not already attached. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_002: [** The `AmqpTwinClient` constructor shall throw `ReferenceError` if the `client` object is falsy. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_008: [** If attaching the sender link fails, the `getTwin` method shall call its callback with the error that caused the failure. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_005: [** The `AmqpDeviceMethodClient` shall inherit from the `EventEmitter` class. **]**
```

Comment:
>AmqpDeviceMethodClien [](start = 45, length = 21)

Testing that I can post a comment.  I can!!  Although one could ask, how did methodClient get through here!